distribution/tools/launchers/src/main/java/org/elasticsearch/tools/launchers/JvmErgonomics.java

```diff
@@ -53,25 +53,57 @@ private JvmErgonomics() {
      */
     static List<String> choose(final List<String> userDefinedJvmOptions) throws InterruptedException, IOException {
         final List<String> ergonomicChoices = new ArrayList<>();
-        final Map<String, Optional<String>> finalJvmOptions = finalJvmOptions(userDefinedJvmOptions);
+        final Map<String, JvmOption> finalJvmOptions = finalJvmOptions(userDefinedJvmOptions);
         final long heapSize = extractHeapSize(finalJvmOptions);
         final long maxDirectMemorySize = extractMaxDirectMemorySize(finalJvmOptions);
         if (maxDirectMemorySize == 0) {
             ergonomicChoices.add("-XX:MaxDirectMemorySize=" + heapSize / 2);
         }
+        final boolean tuneG1GCForSmallHeap = tuneG1GCForSmallHeap(finalJvmOptions, heapSize);
+        final boolean tuneG1GCForLargeHeap = tuneG1GCForLargeHeap(finalJvmOptions, heapSize);
+        if (tuneG1GCForSmallHeap) {
+            ergonomicChoices.add("-XX:G1HeapRegionSize=4m");
+            ergonomicChoices.add("-XX:G1ReservePercent=15");
+            ergonomicChoices.add("-XX:InitiatingHeapOccupancyPercent=45");
+        } else if (tuneG1GCForLargeHeap) {
```

Comment:
Same comment as above, is surprising that if one of them is specified you loose the other. In particular if IHOP is specified, it seems quite dangerous to no longer set `G1ReservePercent`.

Reply:
Thank you for catching this, I updated the pr.